### DISCUSSION TITLE
Why use bounded stores + slices? Why not just use multiple stores?

### DISCUSSION THREAD

**Discussion by williamlmao** [pmndrs/zustand#2347] [Q&A] [answered]
Are the performance implications here, or is it just the benefit of being able to update multiple stores in one action? Thanks!

**Comment by dbritto-dev** [comment, marked as answer]
@williamlmao it depends on your needs if those multiple stores have some deps between each other then makes sense to have one store and multiple slices instead of multiple stores

**Reply by williamlmao** [reply]
Got it, thanks! The documentation seems to suggest that 1 big store is always the way to go, so I was a little confused. It sounds like if i have an encapsulated feature (a global modal state), its totally fine to keep that in a separate store. And I guess if I needed to combine it later on, it would be trivial. Just wanted to make sure I wasn't taking some sort of hit for keeping it separate.

**Comment by ilarramendi** [comment]
I believe this should be added to the documentation; I was also under the impression that multiple stores would have some destructive impact.

**Reply by mordonez-me** [reply]
Yeah me too, in fact I spent hours looking for some real info about this, this is closest thing to an official message about this.

**Reply by dbritto-dev** [reply]
@ilarramendi @mordonez-me actually, we're working on revamping docs

**Reply by Deanfei** [reply]
Me too here. We are going to use zustand in a new project soon. And I am seeking advices and need to make a decision as soon as possible. Would u mind share the new docs if that's possible? All our team is new to zustand and we really need some more detailed docs on some architectural decisions. 

I totally understand that Zustand is unopinionated but some recommendations from the official doc would be really helpful such as is there any destructive impact or performance hit between multiple stores vs multiple slices. Multiple slices are mentioned in the doc. I spent few days to seek for solutions on the naming conflicts without adding another external libraries.

**Comment by dai-shi** [comment]
https://github.com/pmndrs/zustand/discussions/2347#discussioncomment-10170898

> I totally understand that Zustand is unopinionated

😄 

> is there any destructive impact or performance hit between multiple stores vs multiple slices.

Both are really fine, but there are some differences. For example, if you want to use `persist` middleware to store a single item, a single store might be useful. Theoretically, multiple stores are more performant than multiple slices, but it's not meaningfully different in practice.

> the naming conflicts without adding another external libraries.

Technically, you should be able to create a small helper function on your own.

**Reply by Deanfei** [reply]
That's good to know. Thanks!
